v5/core/file_with_metadata_test.go

```diff
@@ -0,0 +1,85 @@
+package core
+
+// (C) Copyright IBM Corp. 2021.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"io/ioutil"
+	"testing"
+)
+
+func TestFileWithMetadataFields(t *testing.T) {
+	data := ioutil.NopCloser(bytes.NewReader([]byte("test")))
+	filename := "test.txt"
+	contentType := "application/octet-stream"
+
+	model := FileWithMetadata{
+		Data: data,
+		Filename: &filename,
+		ContentType: &contentType,
+	}
+
+	assert.NotNil(t, model.Data)
+	assert.NotNil(t, model.Filename)
+	assert.NotNil(t, model.ContentType)
+}
+
+
+func TestNewFileWithMetadata(t *testing.T) {
+	data := ioutil.NopCloser(bytes.NewReader([]byte("test")))
+	model, err := NewFileWithMetadata(data)
+
+	assert.Nil(t, err)
+	myData, ok := model.Data.(io.ReadCloser)
+	assert.True(t, ok)
+	assert.NotNil(t, myData)
+
+	assert.Nil(t, model.Filename)
+	assert.Nil(t, model.ContentType)
+}
+
+func TestUnmarshalFileWithMetadata(t *testing.T) {
+	var err error
+
+	// setup the test by creating a temp file for the unmarshaler to read
+	tempdir := t.TempDir() // this is automatically cleaned up
```

Comment:
Need to code around the fact that TempDir() is not present until Go 1.15, but we currently support Go 1.12+

Reply:
Ah rats, I didn't notice that. I didn't realize I had Go 1.15 locally either but I guess I do. I can go back to what I was doing before I found that function and create my own temp dir